### ISSUE TITLE
REPL is not keyboard accessible

### ISSUE THREAD

**Issue by bpasero** [Microsoft/vscode#2294]
We seem to hide focus in the repl and complex hover when you click with the left mouse, but not with the right mouse. Not sure what the story is here but it is also a problem for accessibility because I cannot access elements using keyboard.

**Comment by isidorn** [comment]
Whenever you left click in the repl we always focus input. Chrome does the same for example.
Right click behavior we do not do anything special.

We should talk about hover and repl accessibility and then figure out what to do here.

**Comment by isidorn** [comment]
@bpasero Ok, so these are two separate issues.
1. Hover is not accessible, Joao had a nice idea that we have a keybinding for showing the hover on a particular token, when the user triggers the hover with this keybinding only then will the hover show focus and then the user can navigate through the elements via keyboard. We have this problem for all our hovers

2. Repl is not accessible, currently the user can just tab in to the repl tree, this way the user can navigate through whole repl history via keyboard. There are some issue here which I can addresss (focused element is not revealed). Not sure what else is missing here

We can also bring this up in our weekly ux sync. Second issue I can try to address right now, but the first one I would wait for our general hover solution.

**Comment by bpasero** [comment]
Yeah starting with 2) seems good, someone should be able to navigate the repl without the mouse. 

Having a keybinding for showing the hover makes sense. But I guess we would need another keybinding to jump into the hover somehow.

**Comment by isidorn** [comment]
Ok I will look at these issue as improve REPL accessibility and will look into that.

That keybinding could automatically focus the hover, so you would not need to jump in. Whenever the hover closes it would give focus back to the editor

**Comment by isidorn** [comment]
@bpasero Ok I was checking out the repl and it actually behaves decently - like all our other trees.
Not sure though if it would be better for the user that when focus is passed to repl the last element gets focus, not the first - since it is a repl.
What do you think?
Since people can navigate the REPL via keyboard I would close this issue and wait for feedback unless you have something on top of your head?

**Comment by bpasero** [comment]
Sounds like a good idea to actually focus the last element when you tab into the tree, would suggest to try it. Otherwise I agree, once focus is in, keyboard navigation works. I filed this bug initially because you cannot click into the tree to focus it, but maybe that is not a big deal since you can do the same with Shift+Tab.

**Comment by isidorn** [comment]
I intentionally disabled click to get focus on element to have a feel of a real console. I can undo that, but I think the current behavior is fine.

Now on first keydown I focus the last element in repl
